clang/lib/CodeGen/CGBuiltin.cpp

```diff
@@ -6011,10 +6011,27 @@ RValue CodeGenFunction::EmitBuiltinExpr(const GlobalDecl GD, unsigned BuiltinID,
   }
   }
 
+  // These will be emitted as Intrinsic later.
+  auto NeedsDeviceOverload = [&](unsigned BuiltinID) {
+    if (getTarget().getTriple().isAMDGCN()) {
+      switch (BuiltinID) {
+      default:
+        return false;
+      case Builtin::BIlogb:
+      case Builtin::BI__builtin_logb:
+      case Builtin::BIscalbn:
+      case Builtin::BI__builtin_scalbn:
+        return true;
+      }
+    }
+    return false;
+  };
```

Comment:
Should not be a lambda, and should not hardcode the triple. and needs documentation, and a fixme to remove it. Can we check if the libcall is supported or something? Really we should have libm

Reply:
There is `llvm::isLibFuncEmittable(const Module *M, const TargetLibraryInfo *TLI, LibFunc TheLibFunc)` but unsure how to access TLI from CGBuiltin's CodeGenFunction::EmitBuiltinExpr.